Gems/AtomLyIntegration/EditorModeFeedback/Code/Source/EditorModeFeedbackSystemComponent.cpp

```diff
@@ -101,66 +102,100 @@ namespace AZ
+        void EditorModeFeedbackSystemComponent::Reflect(AZ::ReflectContext* context)
         {
             if (auto* serialize = azrtti_cast<AZ::SerializeContext*>(context))
             {
-                serialize->Class<EditorEditorModeFeedbackSystemComponent, AzToolsFramework::Components::EditorComponentBase>()
+                serialize->Class<EditorModeFeedbackSystemComponent, AzToolsFramework::Components::EditorComponentBase>()
                     ->Version(0)
                     ;
 
                 if (auto* editContext = serialize->GetEditContext())
                 {
-                    editContext->Class<EditorEditorModeFeedbackSystemComponent>(
+                    editContext->Class<EditorModeFeedbackSystemComponent>(
                         "Editor Mode Feedback System", "Manages discovery of Editor Mode Feedback effects")
                         ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
                         ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System", 0xc94d118b))
                         ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                         ;
                 }
             }
+
+            EditorModeFeatureProcessor::Reflect(context);
         }
 
-        EditorEditorModeFeedbackSystemComponent::~EditorEditorModeFeedbackSystemComponent() = default;
+        EditorModeFeedbackSystemComponent::~EditorModeFeedbackSystemComponent() = default;
 
-        void EditorEditorModeFeedbackSystemComponent::Activate()
+        void EditorModeFeedbackSystemComponent::Activate()
         {
             AzFramework::ApplicationRequests::Bus::Broadcast(
                 [this](AzFramework::ApplicationRequests::Bus::Events* ebus)
                 {
                     m_registeryEnabled = ebus->IsEditorModeFeedbackEnabled();
                 });
 
+            if (!m_registeryEnabled)
+            {
+                return;
+            }
+
             AzToolsFramework::Components::EditorComponentBase::Activate();
             AzToolsFramework::ViewportEditorModeNotificationsBus::Handler::BusConnect(AzToolsFramework::GetEntityContextId());
             AZ::TickBus::Handler::BusConnect();
 
-            if (m_registeryEnabled)
-            {
-                AZ::Interface<EditorModeFeedbackInterface>::Register(this);
-            }
+            AZ::Interface<EditorModeFeedbackInterface>::Register(this);
+            AZ::RPI::FeatureProcessorFactory::Get()->RegisterFeatureProcessor<EditorModeFeatureProcessor>();
+            auto* passSystem = RPI::PassSystemInterface::Get();
+            AZ_Assert(passSystem, "Cannot get the pass system.");
+
+            // Setup handler for load pass templates mappings
+            m_loadTemplatesHandler = RPI::PassSystemInterface::OnReadyLoadTemplatesEvent::Handler(
+                [this]()
+                {
+                    this->LoadPassTemplateMappings();
```

Comment:
uber nit: Unless this is a template the `this->` is redundant, could just be removed

Reply:
Fixed 👍